primitives/src/crh/poseidon/updatable/mod.rs

```diff
@@ -0,0 +1,217 @@
+
+        // Test initializing UpdatablePoseidonHash with personalization is the same as concatenating
+        // to PoseidonHash input the personalization and the padding.
+        let input = MNT4Fr::rand(&mut rng);
+        let samples = 10;
+        for i in 1..=samples{
+            let personalization = vec![MNT4Fr::rand(&mut rng); i];
+
+            let mut hash_input = personalization.clone();
+            let padding = vec![MNT4Fr::zero(); personalization.len() % MNT4753PoseidonParameters::R];
+            hash_input.extend_from_slice(padding.as_slice());
+            hash_input.push(input.clone());
+            let hash_output = MNT4PoseidonHash::evaluate(hash_input.as_slice()).unwrap();
+
+            let updatable_hash_output = {
+                let mut updatable_hash = UpdatableMNT4PoseidonHash::new(Some(personalization.as_slice()));
+                updatable_hash.update(&input);
+                updatable_hash.finalize()
+            };
+            assert_eq!(hash_output,
+                       updatable_hash_output,
+                       "Hashes output with {} elements of personalization must be equal", i
+            );
+        }
+    }
+
+    #[test]
+    fn test_updatable_poseidon_hash_mnt6(){
+        let samples = 100;
+        let mut rng = XorShiftRng::seed_from_u64(1231275789u64);
+
+        for i in 1..=samples {
+            let input = vec![MNT6Fr::rand(&mut rng); i];
+            let hash_output = MNT6PoseidonHash::evaluate(input.as_slice()).unwrap();
+            let updatable_hash_output = {
+                let mut updatable_hash = UpdatableMNT6PoseidonHash::new(None);
+                for input in input.into_iter() {
+                    updatable_hash.update(&input);
+                }
+                updatable_hash.finalize()
+            };
+            assert_eq!(hash_output, updatable_hash_output, "Hashes output with {} inputs must be equal", i);
+        }
+
+        // Test finalize() holding the state and allowing updates in between different calls to it
+        let input = vec![MNT6Fr::rand(&mut rng); 2];
+        let h_out = MNT6PoseidonHash::evaluate(input.as_slice()).unwrap();
+
+        let mut uh =  UpdatableMNT6PoseidonHash::new(None);
+        uh.update(&input[0]);
+        uh.finalize();
+        uh.update(&input[1]);
+        assert_eq!(h_out, uh.finalize());
+
+        //Test finalize() being idempotent
+        assert_eq!(h_out, uh.finalize());
+
+        // Test initializing UpdatablePoseidonHash with personalization is the same as concatenating
+        // to PoseidonHash input the personalization and the padding.
+        let input = MNT6Fr::rand(&mut rng);
+        let samples = 10;
+        for i in 1..=samples{
+            let personalization = vec![MNT6Fr::rand(&mut rng); i];
+
+            let mut hash_input = personalization.clone();
+            let padding = vec![MNT6Fr::zero(); personalization.len() % MNT6753PoseidonParameters::R];
```

Comment:
The same as MNT4.
It should be MNT6753PoseidonParameters::R - personalization.len() % MNT6753PoseidonParameters::R